parser-typechecker/src/Unison/Syntax/TermParser.hs

```diff
@@ -95,7 +95,7 @@ rewriteBlock = do
     rewriteCase = rewriteTermlike "case" DD.rewriteCase
     rewriteType = do
       kw <- quasikeyword "signature"
-      vs <- fromMaybe [] <$> optional (some prefixDefinitionName <* symbolyQuasikeyword ".")
+      vs <- P.try (some prefixDefinitionName <* symbolyQuasikeyword ".") <|> pure []
```

Comment:
This one line fix makes it optional to introduce variables in a `signature` rule. Previously, you'd have to write `signature unused . Optional ==> Optional2` even if the rule didn't need any type variables.